File: 191-number-of-1-bits/number-of-1-bits.cpp
class Solution {
public:
    // Time Complexity: O(log n)
    // Space Complexity: O(1)
    // Approach: To find the hamming weight we keep storing the 
    // least significant set bit until the number n reduces to 0.
    // Use Brian-Kerningham algorithm
    int hammingWeight(uint32_t n) {
        int hamming_wt = 0;

        while(n > 0) {
            hamming_wt++;
            n &= (n - 1);
        }
        return hamming_wt;
    }
};

File: 191-number-of-1-bits/number-of-1-bits.py
class Solution:
    # Time complexity: O(log n)
    # Space complexity: O(1)
    def hammingWeight(self, n: int) -> int:
        hamming_wt = 0

        while n > 0:
            n = n & (n - 1)
            hamming_wt += 1

        return hamming_wt
        